apps/demo/app/custom-ui/[...puckPath]/client.tsx

```diff
@@ -0,0 +1,356 @@
+"use client";
+
+import { Data } from "@/core/types/Config";
+import { Puck } from "@/core/components/Puck";
+import { Render } from "@/core/components/Render";
+import { Button } from "@/core/components/Button";
+import { HeadingAnalyzer } from "@/plugin-heading-analyzer/src/HeadingAnalyzer";
+import config from "../../../config";
+import { useDemoData } from "../../../lib/use-demo-data";
+import { IconButton, usePuck } from "@/core";
+import { ReactNode, useEffect, useRef, useState } from "react";
+import { Drawer } from "@/core/components/Drawer";
+import { ChevronUp, ChevronDown, Globe } from "react-feather";
+
+const CustomHeader = ({ onPublish }: { onPublish: (data: Data) => void }) => {
+  const { appState, dispatch } = usePuck();
+
+  return (
+    <header
+      style={{
+        display: "flex",
+        flexWrap: "wrap",
+        gap: 16,
+        padding: "16px 24px",
+        background: "white",
+        color: "black",
+        alignItems: "center",
+        borderBottom: "1px solid #ddd",
+      }}
+      onClick={() => dispatch({ type: "setUi", ui: { itemSelector: null } })}
+    >
+      <span style={{ fontWeight: 600 }}>Custom UI example </span>
+      <div style={{ marginLeft: "auto", display: "flex", gap: 8 }}>
+        <div>
+          <Button
+            onClick={() => onPublish(appState.data)}
+            icon={<Globe size="14" />}
+          >
+            Publish
+          </Button>
+        </div>
+      </div>
+    </header>
+  );
+};
+
+const Tabs = ({
+  tabs,
+  onTabCollapse,
+  scrollTop,
+}: {
+  tabs: { label: string; body: ReactNode }[];
+  onTabCollapse: () => void;
+  scrollTop: number;
+}) => {
+  const [currentTab, setCurrentTab] = useState(-1);
+  const { appState } = usePuck();
+
+  const currentTabRef = useRef(currentTab);
+
+  useEffect(() => {
+    if (currentTabRef.current !== -1 && appState.ui.itemSelector) {
+      setCurrentTab(1);
+    }
+  }, [appState.ui.itemSelector]);
+
+  useEffect(() => {
+    currentTabRef.current = currentTab;
+  }, [currentTab]);
+
+  useEffect(() => {
+    if (appState.ui.isDragging && currentTab === 1) {
+      setCurrentTab(-1);
+    }
+  }, [currentTab, appState.ui.isDragging]);
+
+  useEffect(() => {
+    if (scrollTop === 0) {
+      setCurrentTab(-1);
+      onTabCollapse();
+    }
+  }, [scrollTop]);
+
+  return (
+    <div
+      onClick={(e) => e.stopPropagation()}
+      style={{
+        background: "#ffffff",
+        pointerEvents: "all",
+        borderTop: "1px solid #ddd",
+        boxShadow: "rgba(140, 152, 164, 0.25) 0px 0px 6px 0px",
+      }}
+    >
+      <div
+        style={{
+          display: "flex",
+          paddingLeft: 16,
+          paddingRight: 16,
+          borderBottom: "1px solid #ddd",
+          overflowX: "auto",
+        }}
+      >
+        {tabs.map((tab, idx) => {
+          const isCurrentTab = currentTab === idx;
+          return (
+            <div
```

Comment:
Switched to button and added `cursor: pointer` to the inline CSS. Still no hover styles because I didn't want to create a stylesheet for the time-being, but this is an improvement.